### PR TITLE
Structure and modules set up and blog posts list component added

### PR DESCRIPTION
**Pull request by DHolovko-CTCo**

### CODE REVIEW

File: src/app/state/blogpost/blogpost.effects.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { map, mergeMap } from 'rxjs/operators';
+import { Post } from 'src/app/blogpost/post';
```
Comment (kaspars01) [inline src/app/state/blogpost/blogpost.effects.ts]: Unused import - usually in the IDE ctrl + alt + o webstorm (alt + shift + o vscode) will automatically optimize imports

File: src/app/state/blogpost/actions/post-api.actions.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Post } from '../../../blogpost/post';
+
+/* NgRx */
+import { createAction, props } from '@ngrx/store';
+
+export const loadPostsSuccess = createAction(
```
Comment (kaspars01) [inline src/app/state/blogpost/actions/post-api.actions.ts]: We used to keep related actions in a single file (and prefix with _action_) :
actionLoadPosts
actionLoadPostsSuccess
actionLoadPostsError

in that manner it is easier to maintain related actions..
and prefix makes autocomplete in IDE simplier - as large projects will have hundreds of actions. 
after that you can type: _aclopo_ or similar part of action and ide will filter out most actions:
this.store.dispatch(_ac_lo_po_ctrl+space

File: src/app/blogpost/post-list/post-list.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<div class="post-list">
+  <div class="post-list--container">
```
Comment (kaspars01) [inline src/app/blogpost/post-list/post-list.component.html]: no need to prefix all classes with parent name
you can use nesting in the css / scss
````
.post-list {
  .container { ... }
  .heading { ... }
  .item { ... }
}
```

File: src/app/blogpost/post-list/post-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+@Component({
+  selector: 'bpt-post-list',
+  templateUrl: './post-list.component.html',
+  styleUrls: ['./post-list.component.scss'],
+  standalone: true,
+  imports: [CommonModule, MatExpansionModule],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PostListComponent implements OnInit {
+  posts$!: Observable<Post[]>;
```
Comment (kaspars01) [inline src/app/blogpost/post-list/post-list.component.ts]: you can simplify code by initializing posts$:
posts$ = this.store.select(getPosts);
and then skip re-assigning in the init

File: src/app/state/blogpost/blogpost.effects.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+export class BlogpostEffects {
+  constructor(
+    private actions$: Actions,
+    private blogpostService: BlogpostService
+  ) {}
+
+  loadPosts$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(PostPageActions.loadPosts),
+      mergeMap(() =>
```
Comment (kaspars01) [inline src/app/state/blogpost/blogpost.effects.ts]: it's better to use switchMap for load related action and mergeMap for save related action.

you can see the difference by calling dispatch twice:

this.store.dispatch(PostPageActions.loadPosts());
this.store.dispatch(PostPageActions.loadPosts());

**Review by kaspars01 — APPROVED** [review]
In general - looks good. 
Treat comments as a suggestions rather than request for code change.

File: src/app/blogpost/post-list/post-list.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="post-list--heading">
+      <h1>Posts</h1>
+    </div>
+
+    <div class="post-list--list">
+      <div *ngIf="posts$ | async">
```
Comment (kaspars01) [inline src/app/blogpost/post-list/post-list.component.html]: it's possible to cache posts
`posts$ | async as posts`
and later re-use 
`*ngFor="let post of posts"`

File: src/app/blogpost/post-list/post-list.component.html (hunk 3 above)
Comment (DHolovko-CTCo) [reply src/app/blogpost/post-list/post-list.component.html]: yeah, in here thought if I would try arranging styles according to BEM later down the track, but maybe not as it seem to be not stacking well with Angular where every component has its own styles. On the other hand could still be good if I'd make a common styles library and leave component-specific styles for fine-tuning and overrides, still thinking about that while waiting on additional info on what practical part of the training is going to be about

File: src/app/state/blogpost/actions/post-api.actions.ts (hunk 2 above)
Comment (DHolovko-CTCo) [reply src/app/state/blogpost/actions/post-api.actions.ts]: so suggestion is to keep actions for APIs and actions for components in the same file, or I am misunderstanding?

Comment (kaspars01) [reply src/app/state/blogpost/actions/post-api.actions.ts]: usually it's one file per feature, but feel free to organize actions more gradually if that suits better

**Pull request merged**